jpa/subsystem/src/main/java/org/jboss/as/jpa/transaction/TransactionUtil.java

```diff
@@ -45,23 +52,51 @@
 
     private static volatile TransactionSynchronizationRegistry transactionSynchronizationRegistry;
     private static volatile TransactionManager transactionManager;
+    private static volatile TransactionListenerRegistry transactionListenerRegistry;
```

Comment:
Static fields should never be used as state storage by subsystems unless there is no other option. The reason is that it leads to leaking and it also prevents the possibility of running more than one container in a VM. Granted the latter is prevented by other code using statics, but we should at least avoid introducing additional ones. Instead a service can be used to hold this common state. 

That said I'm ok with the temporary expansion as long as there is a plan to refactor them out.